WordPress/gutenberg
# Add block-specific commands as contextual suggestions [#53539]

## What?

This PR adds a new context named `block-selection-edit` which is applied to the command palette when a block (or group of blocks) are selected. This allows us to show the block actions in the command palette immediately rather than searching for them.

## Why?
Related to https://github.com/WordPress/gutenberg/issues/53539, closes #53539.


## How?
By adding additional logic to the edit-site and edit-post packages (when context is set) we can apply our new context. I could use assistance improving this to simplify the logic maybe with global context conditions that can be defined in one place rather than duplicating code.


## Testing Instructions
1. Open a post, page, or template.
2. Without selecting a block open up the command palette. You should not see any block action commands (group/ungroup, add before/after, etc).
3. Now select a block or multiple blocks and open up the command palette again. You should see any block action commands for the selected block (group/ungroup, add before/after, etc).





### Testing Instructions for Keyboard


## Screenshots or screencast 
<img width="913" alt="Screenshot 2023-08-27 at 2 17 37 PM" src="https://github.com/WordPress/gutenberg/assets/375788/36ae4683-ec64-4e14-afdf-e6261a04ef79">
